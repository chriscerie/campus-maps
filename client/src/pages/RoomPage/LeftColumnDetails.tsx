import { Link } from 'react-router-dom';
import LocationSection from './LocationSection';
import './LeftColumnDetails.scss';
import { Button } from '@mui/material';

function LeftColumnDetails({
  classroomInfo,
  id,
}: {
  classroomInfo: string;
  id: string;
}) {
  return (
    <div id="classroom-page-left-column">
      <LocationSection title="Description">
        <div className="building-info align-text-left">
          <p style={{ margin: 0 }}>
            This is a description for classroom {classroomInfo}
          </p>
        </div>
      </LocationSection>
    </div>
  );
}

export default LeftColumnDetails;
