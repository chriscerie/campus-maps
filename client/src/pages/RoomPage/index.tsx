/* eslint-disable eqeqeq */
import { Container } from '@mui/material';
import PhotoHeader from './PhotoHeader';
import { useSelector } from 'react-redux';
import { RootState } from '../../reducers';
import { useParams } from 'react-router-dom';
import { useState, useEffect } from 'react';
import LeftColumnDetails from './LeftColumnDetails';
import { getLocationInfo } from '../../api/LocationAPI';
import type { LocationType } from '../../types/LocationType';
import './index.scss';

function RoomPage() {
  const { id, room_id } = useParams<{ id: string; room_id: string }>();
  const [roomInfo, setRoomInfo] = useState<null | { room_name: string }>(null);
  const mapInstance = useSelector((state: RootState) => state.mapInstance.map);

  // Set point
  useEffect(() => {
    if (mapInstance) {
      getLocationInfo(id, mapInstance).then((res) => {
        if (res) {
          // Get room_name that matches room_id
          const room_name = res.rooms.find((room) => room.room_id === room_id);
          if (room_name) {
            setRoomInfo({ room_name: room_name.room_name });
          }
        }
      });
    }
  }, [id, mapInstance, room_id]);

  return (
    <div className="location-page-container">
      <PhotoHeader name={roomInfo?.room_name || ''} id={id} />

      <Container id="location-page-container-inner">
        <LeftColumnDetails classroomInfo={roomInfo?.room_name || ''} id={id} />
      </Container>
    </div>
  );
}

export default RoomPage;
