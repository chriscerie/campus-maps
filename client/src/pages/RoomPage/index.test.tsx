import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../reducers';
import RoomPage from './index';

jest.mock('react-router-dom', () => ({
  __esModule: true,
  useParams: jest.fn().mockReturnValue({
    id: '1',
  }),
}));

describe('LocationPage', () => {
  it('should render successfully', () => {
    const renderer = ShallowRenderer.createRenderer();
    const tree = renderer.render(
      <Provider store={store}>
        <RoomPage />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
