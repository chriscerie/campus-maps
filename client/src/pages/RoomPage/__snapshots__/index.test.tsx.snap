// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationPage should render successfully 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <RoomPage />
</Context.Provider>
`;
